src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -331,9 +332,9 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
             if ((NULL != provname) && (0 != strcmp(provname, hints->fabric_attr->prov_name)))
                 continue;
 
-            /* Check that this provider meets the mimum requirements for the user */
-            if (MPIDI_OFI_ENABLE_DATA && 0ULL == (prov_use->caps & FI_DIRECTED_RECV)) {
-                MPL_DBG_MSG_FMT(MPIDI_CH4_DBG_GENERAL,VERBOSE,(MPL_DBG_FDEST, "Provider doesn't support directed receive (for immediate data)"));
+            /* Check that this provider meets the minimum requirements for the user */
+            if (MPIDI_OFI_ENABLE_DATA && 0ULL == (prov_use->caps & FI_DIRECTED_RECV) && (prov_use->domain_attr->cq_data_size >= 4)) {
```

Comment:
What do you mean about two required capabilities? There's one cap and one attribute here.

Either way, I think it needs more `( )`.